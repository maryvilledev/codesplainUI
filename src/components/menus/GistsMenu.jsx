import React, { Component, PropTypes } from 'react';
import _ from 'lodash';
import { MenuItem } from 'material-ui';
import ArrowDropRight from 'material-ui/svg-icons/navigation-arrow-drop-right';

import CustomPropTypes from '../../util/custom-prop-types';

const makeMenuItems = (gists, onClick) => {
  if (_.isEmpty(gists)) {
    return null;
  }
  return gists.map(({ name, url }) => (
    <MenuItem
      key={name}
      onTouchTap={() => onClick(name, url)}
      primaryText={name}
    />
  ));
};

class GistMenu extends Component {
  constructor(props) {
    super(props);
    this.state = {
      focusState: 'none',
    };
    this.handleOnTouchTap = this.handleOnTouchTap.bind(this);
  }
  componentWillUnmount() {
    this.setState({ focusState: 'none' });
  }

  handleOnTouchTap() {
    this.setState({
      focusState: 'focused',
    });
  }

  render() {
    const { onClick, gists } = this.props;
    const { focusState } = this.state;
    const menuItems = makeMenuItems(gists, onClick);
    return (
      <MenuItem
        disabled={!menuItems}
        focusState={focusState}
        menuItems={menuItems}
        onTouchTap={this.handleOnTouchTap}
        primaryText="Import Gist"
        rightIcon={<ArrowDropRight />}
      />
    );
  }
}

GistMenu.propTypes = {
  onClick: PropTypes.func.isRequired,
  gists: CustomPropTypes.gists,
};

GistMenu.defaultProps = {
  gists: [],
};

export default GistMenu;
