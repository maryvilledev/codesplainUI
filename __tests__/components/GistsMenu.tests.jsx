import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import GistsMenu from '../../src/components/menus/GistsMenu';

const defaultProps = {
  onClick: jest.fn(),
};

describe('<GistsMenu />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = node => shallow(node, { context: { muiTheme } });

  describe('snapshot tests', () => {
    it('matches when no snippets are passed', () => {
      const wrapper = shallowWithContext(
        <GistsMenu
          {...defaultProps}
        />,
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    it('matches when gists are passed', () => {
      const gists = [{ name: 'phonixpersonanatomy.py', url: 'www.rickandmorty100years.com' }];
      const wrapper = shallowWithContext(
        <GistsMenu
          {...defaultProps}
          gists={gists}
        />,
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
