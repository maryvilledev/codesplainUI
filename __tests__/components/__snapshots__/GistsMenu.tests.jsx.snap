exports[`<GistsMenu /> snapshot tests matches when gists are passed 1`] = `
<MenuItem
  anchorOrigin={
    Object {
      "horizontal": "right",
      "vertical": "top",
    }
  }
  checked={false}
  desktop={false}
  disabled={false}
  focusState="none"
  insetChildren={false}
  menuItems={
    Array [
      <MenuItem
        anchorOrigin={
            Object {
                "horizontal": "right",
                "vertical": "top",
              }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        onTouchTap={[Function]}
        primaryText="phonixpersonanatomy.py" />,
    ]
  }
  onTouchTap={[Function]}
  primaryText="Import Gist"
  rightIcon={<NavigationArrowDropRight />} />
`;

exports[`<GistsMenu /> snapshot tests matches when no snippets are passed 1`] = `
<MenuItem
  anchorOrigin={
    Object {
      "horizontal": "right",
      "vertical": "top",
    }
  }
  checked={false}
  desktop={false}
  disabled={true}
  focusState="none"
  insetChildren={false}
  menuItems={null}
  onTouchTap={[Function]}
  primaryText="Import Gist"
  rightIcon={<NavigationArrowDropRight />} />
`;
